src/Pages/Applications/DevtronApps/Details/AppConfigurations/MainContent/DeploymentConfigCompare/DeploymentConfigCompare.tsx

```diff
@@ -0,0 +1,369 @@
+import { useEffect, useMemo } from 'react'
+import { generatePath, useLocation, useRouteMatch } from 'react-router-dom'
+import { GroupBase, OptionsOrGroups } from 'react-select'
+import moment from 'moment'
+
+import {
+    useUrlFilters,
+    SelectPickerOptionType,
+    SelectPickerVariantType,
+    useAsync,
+    ErrorScreenManager,
+    DeploymentConfigDiff,
+    DeploymentConfigDiffProps,
+    useStateFilters,
+} from '@devtron-labs/devtron-fe-common-lib'
+
+import { URLS } from '@Config/routes'
+import { Moment12HourFormat } from '@Config/constants'
+import {
+    AppEnvDeploymentConfigQueryParamsType,
+    AppEnvDeploymentConfigType,
+} from '@Pages/Applications/DevtronApps/service.types'
+import { getAppEnvDeploymentConfig } from '@Pages/Applications/DevtronApps/service'
+import { getOptions } from '@Components/deploymentConfig/service'
+
+import { BASE_CONFIGURATIONS } from '../../AppConfig.constants'
+import {
+    AppEnvDeploymentConfigQueryParams,
+    DeploymentConfigCompareProps,
+    DeploymentConfigParams,
+} from '../../AppConfig.types'
+import {
+    getAppEnvDeploymentConfigList,
+    getDefaultVersionAndPreviousDeploymentOptions,
+    getEnvironmentIdByEnvironmentName,
+    getOptionByValue,
+    getPreviousDeploymentOptionValue,
+    getPreviousDeploymentValue,
+    parseCompareWithSearchParams,
+} from './utils'
+
+export const DeploymentConfigCompare = ({ environments, appName }: DeploymentConfigCompareProps) => {
+    // HOOKS
+    const { path, params } = useRouteMatch<DeploymentConfigParams>()
+    const { appId, resourceType, resourceName, envName } = params
+    const { pathname, search } = useLocation()
+    const { sortBy, sortOrder, handleSorting } = useStateFilters({
+        initialSortKey: 'sort-config',
+    })
+
+    // SEARCH PARAMS
+    const {
+        compareWith,
+        identifierId,
+        pipelineId,
+        compareWithIdentifierId,
+        compareWithPipelineId,
+        compareWithConfigType,
+        configType,
+        chartRefId,
+        updateSearchParams,
+    } = useUrlFilters<never, AppEnvDeploymentConfigQueryParamsType>({
+        parseSearchParams: parseCompareWithSearchParams,
+    })
+
+    // Ensure default query parameters are set if they are not already present
+    useEffect(() => {
+        if ((!chartRefId && !compareWithConfigType) || !configType) {
+            updateSearchParams({
+                compareWithConfigType: compareWithConfigType || AppEnvDeploymentConfigType.PUBLISHED_ONLY,
+                configType: configType || AppEnvDeploymentConfigType.PUBLISHED_ONLY,
+            })
+        }
+    }, [])
```

Comment:
Instead of adding in the effect, can we add the check in the parser instead?

Reply:
can you provide example

Reply:
We can update the method like this i think. Please do ignore any typos and non-optimized code.

Also use AppEnvDeploymentConfigQueryParams for the keys here as well if this is NA

```
export const parseCompareWithSearchParams = (searchParams: URLSearchParams): AppEnvDeploymentConfigQueryParamsType => {
    const identifierId = searchParams.get('identifierId')
    const compareWithIdentifierId = searchParams.get('compareWithIdentifierId')
    const pipelineId = searchParams.get('pipelineId')
    const compareWithPipelineId = searchParams.get('compareWithPipelineId')
    const chartRefId = searchParams.get('chartRefId')
    let compareWithConfigType = searchParams.get(
            'compareWithConfigType',
        ) as AppEnvDeploymentConfigType

    let configType = searchParams.get('configType') as AppEnvDeploymentConfigType

    if ((!chartRefId && !compareWithConfigType) || !configType) {
        compareWithConfigType = compareWithConfigType || AppEnvDeploymentConfigType.PUBLISHED_ONLY
        configType = configType || AppEnvDeploymentConfigType.PUBLISHED_ONLY
    }

    return {
        [AppEnvDeploymentConfigQueryParams.CONFIG_TYPE]: configType,
        [AppEnvDeploymentConfigQueryParams.COMPARE_WITH]: searchParams.get('compareWith'),
        [AppEnvDeploymentConfigQueryParams.COMPARE_WITH_CONFIG_TYPE]: compareWithConfigType,
        [AppEnvDeploymentConfigQueryParams.IDENTIFIER_ID]: identifierId ? parseInt(identifierId, 10) : null,
        [AppEnvDeploymentConfigQueryParams.PIPELINE_ID]: pipelineId ? parseInt(pipelineId, 10) : null,
        [AppEnvDeploymentConfigQueryParams.COMPARE_WITH_IDENTIFIER_ID]: compareWithIdentifierId
            ? parseInt(compareWithIdentifierId, 10)
            : null,
        [AppEnvDeploymentConfigQueryParams.COMPARE_WITH_PIPELINE_ID]: compareWithPipelineId
            ? parseInt(compareWithPipelineId, 10)
            : null,
        [AppEnvDeploymentConfigQueryParams.CHART_REF_ID]: chartRefId ? parseInt(chartRefId, 10) : null,
    }
}
```